vertx-junit5-web-client/src/main/java/io/vertx/junit5/web/WebClientParameterProvider.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2020 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.vertx.junit5.web;
+
+import io.vertx.core.Vertx;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.WebClientOptions;
+import io.vertx.junit5.ParameterClosingConsumer;
+import io.vertx.junit5.ScopedObject;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxExtensionParameterProvider;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.platform.commons.util.AnnotationUtils;
+import org.junit.platform.commons.util.ReflectionUtils;
+
+import java.lang.reflect.Field;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+public class WebClientParameterProvider implements VertxExtensionParameterProvider<WebClient> {
+
+  @Override
+  public Class<WebClient> type() {
+    return WebClient.class;
+  }
+
+  @Override
+  public String key() {
+    return "WebClient";
+  }
+
+  @Override
+  public WebClient newInstance(ExtensionContext extensionContext, ParameterContext parameterContext) {
+    ExtensionContext.Store store = extensionContext.getStore(ExtensionContext.Namespace.GLOBAL);
+    ScopedObject scopedObject = store.get(VertxExtension.VERTX_INSTANCE_KEY, ScopedObject.class);
+    Objects.requireNonNull(scopedObject, "A Vertx instance must exist, try adding the Vertx parameter as the first method argument");
+    Vertx vertx = (Vertx) scopedObject.get();
```

Comment:
A couple of questions here:

* How should the rxified version of WebClient be injected here?
  * Where should live the logic to do it? In a separate `ParameterProvider`? And what package should host this `ParameterProvider`? Another separate project (like `vertx-web-client-rx-junit5`)?
  * This `ParameterProvider` should try to discover the rxified `Vertx` right? But then this cast could break if you mix normal vertx types and rxified ones... What if the extension context stores always the original types, while when it's needed it creates a rxified wrap on flight? Does it makes sense?
* Does it makes sense to abstract the `Vertx` instance retrieval in a separate static method? Since this will be a common task for any `ParameterProvider`, a static method should avoid to rewrite this boilerplate